Agrega lógica para validar campos obligatorios

gisell-royg-itti opened the pull request:
asepy-14 Agrega lógica para validar campos obligatorios del formulario
de registro.
Actualmente no se realiza el control, siendo los campos obligatorios:
nombre, apellido, fecha de nacimiento, documento, celular, email y ruc.

gisell-royg-itti committed:
Agrega lógica para validar campos obligatorios

asepy-14 Agrega lógica para validar campos obligatorios del formulario
de registro.
Actualmente no se realiza el control, siendo los campos obligatorios:
nombre, apellido, fecha de nacimiento, documento, celular, email y ruc.

gisell-royg-itti committed:
Agrega validacion y reset de campos obligatorios

asepy-14 Agrega validacion y reset de campos obligatorios para el
formulario de Registro.

gisell-royg-itti committed:
Agrega validacion de campos al editar datos

asepy-14 Agrega validación de campos obligatorios al momento de editar
el registro de un socio.

gisell-royg-itti committed:
Soluciona conflictos con la rama master

gisell-royg-itti commented:
@rodrigojv Hola Rodri! Los cambios de validación de campos se hicieron sobre el RegisterForm y sobre el EditModal.

rodrigojv commented:
Hola @gigiroyg . Estuve revisando el PR y creo que podemos simplificar la validación si nos apoyamos en Formik, sumando otra librería que se llama [Yup](https://github.com/jquense/yup) para validación.

Te diría que pruebes instalar yup `npm install yup`

Luego por ejemplo en el form de EditModal.

```
import { FormErrorMessage } from '@chakra-ui/react'

import * as Yup from "yup"`
```
En tu componente definis un esquema de validación con Yup.

```jsx
const MemberEditSchema = Yup.object().shape({
    name: Yup.string().required('Nombre es requerido'),
    email: Yup.string().email('email inválido').required('Email es requerido')
   // otros campos
  });
```

En la definición de tu formulario:

```jsx
<Formik
              initialValues={values}
              validationSchema={MemberEditSchema} // acá se conecta Yup con Formik
              onSubmit={(values) => handleSubmit(values)}> // se le llama acá directo y recibe los values actualizados
```

Y para mostrar los errores de validación con mensajito de error y resaltado en rojo:

```jsx
      <Field name="name">
                        {({ field, form }) => (
                          <FormControl
                            id={'name'}
                            isInvalid={form.errors.name && form.touched.name}>
                            <FormLabel>Nombres</FormLabel>
                            <Input {...field} id="name" isRequired={true} />
                            <FormErrorMessage>
                              {form.errors.name}
                            </FormErrorMessage>
                          </FormControl>
                        )}
                      </Field>
```

Con estos cambios mencionados ya no es necesario recurrir a `updateValue`, `checkFields`, `isValid`, etc.

rodrigojv commented on components/members/RegisterForm.css:
```diff
@@ -0,0 +1,9 @@
+.invalidField {
```
En teoría no es necesario si se usa `isInvalid` del FormControl.

rodrigojv commented on components/members/RegisterForm.css:
```diff
@@ -0,0 +1,9 @@
+    border-radius: 0.375rem;
+}
+
+.react-datepicker__input-container {
+    padding-right: 0px;
+    padding-left: 0px;
+}
```
Esto se puede colocar dentro del archivo `react-datepicker.css`

gisell-royg-itti commented:
@rodrigojv Genial! Gracias Rodri! Agrego esa libreria y ajusto las demás cosas.

gisell-royg-itti committed:
Modifica lógica al validar campos al editar datos de un socio

gisell-royg-itti committed:
Modifica libreria de formulario de registro

asepy-14 Se utiliza la libreria de Formik para el formulario de registro
asi como el formulario para editar un registro.

gisell-royg-itti committed:
Elimina clase css que ya no se utiliza

rodrigojv commented:
De lujo @gigiroyg! 🔥

Ya se puede mezclar?

gisell-royg-itti commented:
@rodrigojv 🙌 Todavia no se puede mezclar. El toast no está funcionando en el formulario de registro y estoy verificando eso.

rodrigojv commented:
Sí, me di cuenta de eso. Ya tengo una solución en la rama que estoy trabajando actualmente.

rodrigojv commented:
Ya subí a master una solución al issue del toast @gigiroyg  🙆‍♀️

gisell-royg-itti committed:
Soluciona conflictos con la rama master

asepy-14 Agrega select de departamentos y ciudades en el modal de Editar
